src/index.js

```diff
@@ -17,147 +17,19 @@ var cronofy = function(config){
-  }
-
-  this.deleteNotificationChannel = function(){
-    var details = parseArguments(arguments, ["access_token"]);
-
-    return httpDelete('/v1/channels/' + details.options.channel_id, details.options, details.callback, ['access_token', 'channel_id']);
-  }
-
-  this.elevatedPermissions = function(){
-    var details = parseArguments(arguments, ["access_token"]);
-
-    return httpPost('/v1/permissions', details.options, details.callback);
-  }
-
-  this.freeBusy = function(){
-    var details = parseArguments(arguments, ["access_token"]);
-
-    return httpGet(details.options.next_page || '/v1/free_busy', details.options, details.callback);
-  }
-
-  this.listCalendars = function(){
-    var details = parseArguments(arguments, ["access_token"]);
-
-    return httpGet('/v1/calendars', details.options, details.callback);
-  }
-
-  this.listNotificationChannels = function(){
-    var details = parseArguments(arguments, ["access_token"]);
-
-    return httpGet('/v1/channels', details.options, details.callback);
-  }
-
-  this.profileInformation = function(){
-    var details = parseArguments(arguments, ["access_token"]);
-
-    return httpGet('/v1/profiles', details.options, details.callback);
-  }
-
-  this.readEvents = function(){
-    var details = parseArguments(arguments, ["access_token"]);
-
-    return httpGet(details.options.next_page || '/v1/events', details.options, details.callback);
-  }
-
-  this.refreshAccessToken = function(){
-    var details = parseArguments(arguments, ["client_id", "client_secret", "refresh_token"]);
-
-    details.options.grant_type = "refresh_token";
-
-    return httpPost('/oauth/token', details.options).then(tap(function(token){
-      config.access_token = token.access_token;
-      config.refresh_token = token.refresh_token;
-
-
-      if(details.callback){
-        details.callback(null, token);
-      }
-    }), details.callback);
-  }
-
-  this.requestAccessToken = function(){
-    var details = parseArguments(arguments, ["client_id", "client_secret", "refresh_token"]);
-
-    details.options.grant_type = "authorization_code";
-
-    return httpPost('/oauth/token', details.options).then(tap(function(token){
-      config.access_token = token.access_token;
-      config.refresh_token = token.refresh_token;
-
-      if(details.callback){
-        details.callback(null, token);
-      }
-    }), details.callback);
-  }
-
-  this.revokeAuthorization = function(){
-    var details = parseArguments(arguments, ["client_id", "client_secret", "refresh_token"]);
-
-    return httpPost('/oauth/token/revoke', details.options).then(tap(function(){
-      delete config.access_token;
-      delete config.refresh_token;
-
-      if(details.callback){
-        details.callback();
-      }
-    }), details.callback);
-  }
-
   var urls = {
     api: 'http://local' + (config.dataCenter ? '-' + config.dataCenter : '') + '.cronofy.com'
```

Comment:
If you put this on `this` you can move the `httpGet/Post/Delete` methods to the prototype as well.
Also I would prefix them with `_` to let people know they're not public methods.

If you can pass the `urls.api` to the `httpCall` function you can even move that off the prototype to just be an internal function. (my own personal rule is to put as little as needed on the "public" object)